### PR TITLE
#1133 - add approve reimbursement request endpoint

### LINKED ISSUE #1133: Finance - Approve Endpoint
### Description

we need an endpoint for to approve the reimbursement request from the finance team

### Acceptance Criteria

- [ ] New Post Receiver in the reimbursement routes file
  - Receiver will not have a body, calls controller function
  - Route will be /reimbursement-requests/:id/approve
- [ ] New Controller function called approveReimbursementRequest in the reimbursement controller file
  - Will parse the id out of the query 
  - Will get the current user
  - Calls Service function
- [ ] New service function called approveReimbursementRequests in reimbursement services filea
  - Function will throw if user is not a member of the finance team
  - Function should add a Reimbursement Status with the type pending advisor 

- [ ] Unit Tests
### Proposed Solution

^ 

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by RChandler234**
## Changes

Add Approve Reimbursement Request Endpoint

## Notes

Not sure how to do tests since we're checking an environment variable when we check that a user is on the finance team (seems like that's why we don't have unit tests for send pending advisor list or add sabo number as well). Will look into this more later

## To Do

_Any remaining things that need to get done_

- [ ] tests?

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [ ] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1133  (issue #)

### CODE REVIEW

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Looks prety good a couple small things that i noticed just browsing through the code

File: src/backend/src/controllers/reimbursement-requests.controllers.ts [hunk 1]
```diff
@@ -126,4 +126,15 @@ export default class ReimbursementRequestsController {
       next(error);
     }
   }
+
+  static async approveReimbursementRequest(req: Request, res: Response, next: NextFunction) {
+    try {
+      const { requestId } = req.params;
+      const user = await getCurrentUser(res);
+      const reimbursementStatus = await ReimbursementRequestService.approveReimbursementRequest(requestId, user);
```
Comment (Peyton-McKee) [inline src/backend/src/controllers/reimbursement-requests.controllers.ts:134]: you want to return the status not the request?

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 2]
```diff
@@ -95,7 +98,7 @@ export default class ReimbursementRequestService {
         receiptPictures,
         expenseTypeId: expenseType.expenseTypeId,
         totalCost,
-        reimbursementsStatuses: {
+        reimbursementStatuses: {
           create: {
             type: 'PENDING_FINANCE',
```
Comment (Peyton-McKee) [inline src/backend/src/services/reimbursement-requests.services.ts:103]: Cant we do ReimbursementStatuses.Pending_Finance?

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 3]
```diff
@@ -327,4 +330,47 @@ export default class ReimbursementRequestService {
 
     return reimbursementRequests.map(reimbursementRequestTransformer);
   }
+
+  /**
+   * Adds a reimbursement status with type sabo submitted to the given reimbursement request
+   *
+   * @param reimbursementRequestId the id of the reimbursement request to approve
+   * @param submitter the user who is approving the reimbursement request
+   * @returns the reimbursement request with the sabo number
```
Comment (Peyton-McKee) [inline src/backend/src/services/reimbursement-requests.services.ts:339]: this conflicts with what your doing

File: src/backend/src/services/reimbursement-requests.services.ts [hunk 4]
```diff
@@ -327,4 +330,47 @@ export default class ReimbursementRequestService {
...
+
+    if (
+      reimbursementRequest.reimbursementStatuses.some((status) => status.type === ReimbursementStatusType.SABO_SUBMITTED)
+    ) {
+      throw new HttpException(400, 'This reimbursement request has already been approved');
+    }
+
+    const reimbursementStatus = await prisma.reimbursement_Status.create({
+      data: {
+        type: 'SABO_SUBMITTED',
```
Comment (Peyton-McKee) [inline src/backend/src/services/reimbursement-requests.services.ts:365]: can we ReimbursementStatsusTpe.Sabo_Submitted here?

File: src/backend/tests/reimbursement-requests.test.ts [hunk 5]
```diff
@@ -259,4 +262,47 @@ describe('Reimbursement Requests', () => {
...
+    });
+
+    test('Approve Reimbursement Request fails if the request has already been approved', async () => {
+      jest.spyOn(prisma.team, 'findUnique').mockResolvedValue(primsaTeam2);
+      jest.spyOn(prisma.reimbursement_Request, 'findUnique').mockResolvedValue(prismaGiveMeMyMoney2);
+
+      await expect(
+        ReimbursementRequestService.approveReimbursementRequest(prismaGiveMeMyMoney2.reimbursementRequestId, alfred)
+      ).rejects.toThrow(new HttpException(400, 'This reimbursement request has already been approved'));
+    });
```
Comment (Peyton-McKee) [inline src/backend/tests/reimbursement-requests.test.ts:306]: successful test?

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
couple more things

File: src/backend/src/prisma/schema.prisma [hunk 6]
```diff
@@ -349,7 +349,7 @@ model Reimbursement_Request {
   dateCreated            DateTime                @default(now())
   dateDeleted            DateTime?
   dateOfExpense          DateTime
-  reimbursementsStatuses Reimbursement_Status[]
+  reimbursementStatuses  Reimbursement_Status[]
```
Comment (anthonybernardi) [inline src/backend/src/prisma/schema.prisma:352]: the sql file says renamed without changes but this should probs cause a change right

**Review by anthonybernardi — COMMENTED** [review]
awesome

**Pull request merged**